core/src/main/java/org/mule/runtime/core/message/PartAttributes.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.message;
+
+import static java.util.Objects.requireNonNull;
+
+import org.mule.runtime.api.message.MultiPartPayload;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.Map;
+
+/**
+ * Representation of metadata associated to a part in a {@link MultiPartPayload}.
+ *
+ * @since 4.0
+ */
+public class PartAttributes extends BaseAttributes
+{
+    private static final long serialVersionUID = -4718443205714605260L;
+
+    private Map<String, LinkedList<String>> headers;
+
+    private String name;
+
+    private String fileName;
+
+    private long size;
+
+    /**
+     * Builds a new instance of this attributes with the given parameters.
+     * 
+     * @param name The name of the part.
+     * @param fileName The name of the file of the attachment, or {@code null} if it wasn't provided.
+     * @param size The size in bytes of the attachment, or -1 if unknown or undetermined.
+     * @param headers The headers relative to the attachment part.
+     */
+    public PartAttributes(String name,
+            String fileName,
+            long size,
+            Map<String, LinkedList<String>> headers)
+    {
+        requireNonNull(name);
+        this.name = name;
+        this.fileName = fileName;
+        this.size = size;
+        this.headers = ImmutableMap.copyOf(headers);
```

Comment:
Do we want to use static imports for copy() and emptyMap() in this class? We do in most places...